ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala

```diff
@@ -79,35 +81,34 @@ private[platform] object PaginatingAsyncStream {
       .flatMapConcat(Source(_))
   }
 
-  case class IdPaginationState(startOffset: Long, pageSize: Int)
+  case class IdPaginationState(fromIdExclusive: Long, pageSize: Int)
 
-  // TODO etq: Unify with com.daml.platform.store.dao.events.FilterTableACSReader.idSource
   def streamIdsFromSeekPagination(
-      pageConfig: IdQueryConfiguration,
-      pageBufferSize: Int,
-      initialStartOffset: Long,
+      idPageSizing: IdPageSizing,
+      idPageBufferSize: Int,
+      initialFromIdExclusive: Long,
   )(
       fetchPage: IdPaginationState => Future[Vector[Long]]
   ): Source[Long, NotUsed] = {
+    // TODO etq: Make sure this requirement is documented in the config
+    assert(idPageBufferSize > 0)
+    val initialState = IdPaginationState(
+      fromIdExclusive = initialFromIdExclusive,
+      pageSize = idPageSizing.minPageSize,
+    )
     Source
-      .unfoldAsync[IdPaginationState, Vector[Long]](
-        IdPaginationState(
-          startOffset = initialStartOffset,
-          pageSize = pageConfig.minPageSize,
-        ): IdPaginationState
-      ) { state: IdPaginationState =>
+      .unfoldAsync[IdPaginationState, Vector[Long]](initialState) { state =>
         fetchPage(state).map {
           case empty if empty.isEmpty => None
-          case nonEmpty: Vector[Long] =>
-            val nextPageStartOffset = nonEmpty.last
-            val newState = IdPaginationState(
-              startOffset = nextPageStartOffset,
-              pageSize = Math.min(state.pageSize * 4, pageConfig.maxPageSize),
+          case ids =>
+            val nextState = IdPaginationState(
+              fromIdExclusive = ids.last,
+              pageSize = Math.min(state.pageSize * 4, idPageSizing.maxPageSize),
```

Comment:
I am wondering, what is 4?

Reply:
We fetch event ids in pages starting with a very small page size for low latency and then transition into larger page sizes for throughput.
4 is just a multiplier of the page size we use to do that.